examples/routing.js

```diff
@@ -0,0 +1,55 @@
+var plastiq = require('..');
+var h = plastiq.html;
+var router = require('../router');
+
+var objects = [
+  { name: 'one', id: 1 },
+  { name: 'two', id: 2 },
+  { name: 'three', id: 3 },
+  { name: 'four', id: 4 }
+];
+
+function render(model) {
+  return h('div',
+    h('.header', 'Hi ', model.name),
+    router(
+      router.page('/', function () {
+        return h('div',
+          h('h1', 'root'),
+          router.link('/objects', 'objects')
+        );
+      }),
+      router.page('/objects', function () {
+        return new Promise(function (result) {
+          setTimeout(function () {
+            result(objects);
+          }, 300);
+        });
+      }, function (objects) {
+        return h('div',
+          h('h1', 'objects'),
+          h('ul',
+            objects.map(function (o) {
+              return router.link('/objects/' + o.id, h('li', o.name));
```

Comment:
Yeah I agree, too surprising, annoying in some cases.

But I do like the idea of routing anchors looking pretty similar to regular anchors. This may be less funky:

h('a', { route: ['/objects/:id', o.id] }, ...)